dash-pipeline/bmv2/dash_parser.p4

```diff
@@ -29,14 +29,49 @@ parser dash_parser(
     )
 {
     state start {
+        // By default, packet is REGULAR from EXTERNAL
+        hd.packet_meta.setValid();
+        hd.packet_meta.packet_source = dash_packet_source_t.EXTERNAL;
+        hd.packet_meta.packet_type = dash_packet_type_t.REGULAR;
+        hd.packet_meta.packet_subtype = dash_packet_subtype_t.NONE;
+        hd.packet_meta.length = PACKET_META_HDR_SIZE;
+
         packet.extract(hd.u0_ethernet);
         transition select(hd.u0_ethernet.ether_type) {
             IPV4_ETHTYPE:  parse_u0_ipv4;
             IPV6_ETHTYPE:  parse_u0_ipv6;
+            DASH_ETHTYPE:  parse_dash;
             default: accept;
         }
     }
 
+    state parse_dash {
+        packet.extract(hd.packet_meta);
+        if (hd.packet_meta.packet_subtype != dash_packet_subtype_t.NONE) {
```

Comment:
will be better to check packet_subtype against the flow_xxx instead of NONE. it will be more robust and more readable.

Reply:
Fixed